src/swagger/specGenerator3.ts

```diff
@@ -94,10 +94,18 @@ export class SpecGenerator3 extends SpecGenerator {
   private buildServers() {
     const basePath = normalisePath(this.config.basePath as string, '/', undefined, false);
     const scheme = this.config.schemes ? this.config.schemes[0] : 'https';
-    const host = this.config.host || 'localhost:3000';
+
+    let url;
+    if (this.config.host === null) {
+      url = basePath;
+    } else {
+      const host = this.config.host || 'localhost:3000';
+      url = `${scheme}://${host}${basePath}`;
+    }
+
```

Comment:
@jreusch please forgive the delay. This looks good. Please do it for version 2 of the spec too. We have to continue to support Swagger v2 as well as OpenAPI (i.e. Swagger v3 which you made the changes for).

Reply:
hey hey, i checked this back then but totally forgot to mention:
actually version 2 already seems to behave how we'd like it to have for tsoa v3.

e.g.
defining only basePath in tsoa.json and generating  specs results in
for  v2:
`swagger.json 
"basePath": "/v1",
(host missing)
`
for v3:
`swagger.json
"url": "https://localhost:3000/v1"
`

defining the host as example.com and generating the spec:
v2:
`
"basePath": "/v1",
"host": "example.com"
`
v3
`
 "url": "https://example.com/v1"
`

so currently you already get _different_ results depending if you generate a v2 or v3 spec...

Reply:
That’s really unfortunate, but I don’t think it changes our plan if releasing this change, then the breaking version later.